File: regexlib/2021-5-15/javascript_test_file/regexlib_6858.js
/* 6858
 * \b[\w]+[\w.-][\w]+@[\w]+[\w.-]\.[\w]{2,4}\b
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"11"*5000+"!1 _SLQ_2"
 */
var REGEX = /\b[\w]+[\w.-][\w]+@[\w]+[\w.-]\.[\w]{2,4}\b/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '11'.repeat(i*10000) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_6858.py
# 6858
# \b[\w]+[\w.-][\w]+@[\w]+[\w.-]\.[\w]{2,4}\b
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"11"*5000+"!1 _SLQ_2"

import re
from time import perf_counter

regex = """\b[\w]+[\w.-][\w]+@[\w]+[\w.-]\.[\w]{2,4}\b"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "11" * i * 10000 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")